src/core/featuremodel.cpp

```diff
@@ -341,9 +341,20 @@ bool FeatureModel::save()
       {
         QgsFeature modifiedFeature;
         if ( mLayer->getFeatures( QgsFeatureRequest().setFilterFid( mFeature.id() ) ).nextFeature( modifiedFeature ) )
-          setFeature( modifiedFeature );
+        {
+          if ( modifiedFeature != mFeature )
+          {
+            setFeature( modifiedFeature );
+          }
+          else
+          {
+            emit featureUpdated();
```

Comment:
In most scenarios it's

```
if ( new != old ) 
{
  update();
  emitChanged();
}
```
what's different here?

Reply:
OK, so most of the time, "new" (i.e. the fetched feature from the provider after having updated said feature) will be equal to "old" ( the updated feature sent to the provider). For a small subset of feature updates, those which the provider subsequently modify on update, we end up calling setFeature( modifiedFeature[FromProvider] ) which emits a featureChanged signal.

But in most cases, when new == old, we had no way to connect to a signal to be aware of the feature having been updated. Hence the new featureUpdated signal.